### PR TITLE
[c2][decoder] Fixed some issues when oneVPL returns resolution change

### PR DESCRIPTION
**Pull request by zhangyichix**
case: widevineH264AdaptiveWithRendererDisablingV18

1. prevent deleting bs which in use.
2. when oneVPL returns an error of resolution change, it is not necessarily the width or height has changed. however, we reinit the codec.
3. when deleting a work if it is not found, do not return a fatal error directly.

Tracked-On: OAM-104374
Signed-off-by: zhangyichix <yichix.zhang@intel.com>

### CODE REVIEW

**Review by TianmiChen — APPROVED** [review]

File: c2_components/src/mfx_c2_decoder_component.cpp [hunk 1]
```diff
@@ -1041,20 +1041,22 @@ mfxStatus MfxC2DecoderComponent::HandleFormatChange()
...
-                m_mfxVideoParams.mfx.FrameInfo.Height != m_uMaxHeight) {
-                need_realloc = true;
-            } else if (m_uOutputDelay < decRequest.NumFrameSuggested){
+        // mfxFrameAllocRequest decRequest = {};
+        // mfx_res = m_mfxDecoder->QueryIOSurf(&m_mfxVideoParams, &decRequest);
+        // if (MFX_ERR_NONE == mfx_res) {
+            // if (m_mfxVideoParams.mfx.FrameInfo.Width != m_uMaxWidth ||
+            //     m_mfxVideoParams.mfx.FrameInfo.Height != m_uMaxHeight) {
+            //     need_realloc = true;
+            // } else if (m_uOutputDelay < decRequest.NumFrameSuggested){
```
Comment (dyang23) [inline c2_components/src/mfx_c2_decoder_component.cpp:1052]: I would suggest not use // to mask block code, please use #if 0 if this code may add back later, or remove them directly.

**Review by dyang23 — COMMENTED** [review]
Do you mean is resolution change event got, for oneVPL the resolution may not changed, so the older logic of check resolution change may not work anymore, and we just run the reset logic without check width/height again?

**Comment by zhangyichix** [comment]
> Do you mean is resolution change event got, for oneVPL the resolution may not changed, so the older logic of check resolution change may not work anymore, and we just run the reset logic without check width/height again?

yes.

File: c2_components/src/mfx_c2_decoder_component.cpp (hunk 1 above)
Comment (zhangyichix) [reply c2_components/src/mfx_c2_decoder_component.cpp:1052]: ok, I'll detele it.

**Review by dyang23 — APPROVED** [review]

**Pull request merged**